src/test/java/org/opensearch/knn/index/memory/breaker/NativeMemoryCircuitBreakerServiceTests.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.memory.breaker;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.SneakyThrows;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.unit.ByteSizeUnit;
+import org.opensearch.common.unit.ByteSizeValue;
+import org.opensearch.knn.KNNTestCase;
+import org.opensearch.knn.index.KNNSettings;
+import org.opensearch.knn.plugin.stats.StatNames;
+import org.opensearch.knn.plugin.transport.KNNStatsAction;
+import org.opensearch.knn.plugin.transport.KNNStatsNodeResponse;
+import org.opensearch.knn.plugin.transport.KNNStatsRequest;
+import org.opensearch.knn.plugin.transport.KNNStatsResponse;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.opensearch.knn.index.KNNSettings.KNN_CIRCUIT_BREAKER_TRIGGERED;
+import static org.opensearch.knn.index.KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_ENABLED;
+import static org.opensearch.knn.index.KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_LIMIT;
+
+public class NativeMemoryCircuitBreakerServiceTests extends KNNTestCase {
+
+    public void testSetCircuitBreaker() {
```

Comment:
see if you can make test naming consistent in the file, as test_whenABC_thenXYZ

Reply:
Will update.